accumulator/transform.go

```diff
@@ -64,6 +64,43 @@ func remTrans2(dels []uint64, numLeaves uint64, forestRows uint8) [][]arrow {
 	return swaps
 }
 
+// makeCollapse determines whether a collapse should take place given a list
+// of nodes to delete and misc info. If the collapse is present, it consists
+// of an single arrow (a source index and a destination index).
+// A collapse is a deferred swap, which is later made into an actual swap in
```

Comment:
i think explaining what a collapse is in more detail would be nice. the paper also makes no mention of them. Maybe something like: "A collapse is a deferred swap, that swaps a root or the sibling of a leftover deletion to its new position in the modified tree. A collapse can not directly be appended to the regular swaps because the collapse destination changes with upper row swaps. swapCollapses turns the collapse into a regular swap by computing a new destination position so that after upper row swaps the node is located at the collapse destination."

Reply:
Just a note but a "collapse" is called "Root" in the paper. It's outlined in section 7.4

Reply:
hm not entirely, the `!delRemains && rootPresent` case and `swapCollapses` are not in the paper but maybe those are just implementation details

Reply:
Ok, I adopted your comment verbatim.